articles/fleetd-authentication.md

```diff
@@ -0,0 +1,101 @@
+# Fleetd authentication
+
+Fleetd is the agent that runs on the endpoints and communicates with Fleet to periodically send device data and to execute management tasks.
+
+The fleetd agent contains three components:
+- [orbit](https://github.com/fleetdm/fleet/tree/main/orbit): This is the main daemon/service that starts all components on the endpoint.
+- [osquery](https://github.com/osquery/osquery): Provides the endpoint with SQL-powered instrumentation, monitoring, and analytics.
+- [Fleet desktop](https://fleetdm.com/guides/fleet-desktop#basic-article): This is the tray icon app to interface with users of the (Desktop) endpoint. This component can be disabled/missing on some endpoints (e.g. for servers without GUI).
+
+This document aims to explain how all three components authenticate to Fleet servers.
+
+## TLS
+
+All fleetd components communicate to Fleet using HTTPS.
+
+> Fleetd also supports mTLS, more on this below.
+
+## Enrollment to Fleet
+
+Teams in Fleet have an "enroll secret" that is used to authenticate endpoints during enrollment. 
+
+Enroll secrets are generated randomly by Fleet (if not provided) or can be defined by the admin.
+The enroll secrets ramdonly generated by Fleet are 24 bytes long, base64 encoded.
+
+### Fleetd package generation
+
+When generating a fleetd package using `fleetctl`, the admin provides an "enroll secret":
+```sh
+fleetctl package --type=pkg \
+    --fleet-desktop \
+    --fleet-url=... \
+    --enroll-secret=...
+```
+
+The "Fleet URL" and "Enroll secret" are embedded into the generated package.
```

Comment:
Maybe beneficial to link to https://fleetdm.com/guides/enroll-hosts#basic-article to make it easy for people to find more info?